prow/pod-utils/clone/clone_test.go

```diff
@@ -51,3 +53,280 @@ func TestPathForRefs(t *testing.T) {
 		}
 	}
 }
+
+func TestCommandsForRefs(t *testing.T) {
+	var testCases = []struct {
+		name                                       string
+		refs                                       *kube.Refs
+		dir, gitUserName, gitUserEmail, cookiePath string
+		env                                        []string
+		expected                                   []cloneCommand
+	}{
+		{
+			name: "simplest case, minimal refs",
+			refs: &kube.Refs{
+				Org:     "org",
+				Repo:    "repo",
+				BaseRef: "master",
+			},
+			dir: "/go",
+			expected: []cloneCommand{
```

Comment:
Rather than checking the exact list of all commands these tests might be easier to read if we replace `expected` with `required` (aka commands we want to see) and `unexpected` (commands we don't want to see).

This would make it easier to understand that we're trying to validate that when the refs: includes a CloneURI we want it to change the git fetch command, but don't really care about the others.

Reply:
I guess we could do both, get the best of both worlds

Reply:
Hmm on second thought, no -- I think all of these are order dependent. Even the `git config` stuff needs to happen before others to be useful.